k256/src/ecdsa/sign.rs

```diff
@@ -104,10 +118,19 @@ where
 
 impl<D> DigestSigner<D, recoverable::Signature> for SigningKey
 where
-    D: BlockInput + FixedOutput<OutputSize = U32> + Clone + Default + Reset + Update,
+    D: CoreProxy + Digest + OutputSizeUser<OutputSize = U32>,
+    D::Core: BlockSizeUser
+        + BufferKindUser<BufferKind = Eager>
+        + Clone
+        + Default
+        + FixedOutputCore
+        + HashMarker
+        + OutputSizeUser<OutputSize = D::OutputSize>,
+    <D::Core as BlockSizeUser>::BlockSize: IsLess<typenum::U256>,
+    Le<<D::Core as BlockSizeUser>::BlockSize, typenum::U256>: NonZero,
 {
     fn try_sign_digest(&self, msg_digest: D) -> Result<recoverable::Signature, Error> {
-        let z = <Scalar as Reduce<U256>>::from_be_bytes_reduced(msg_digest.finalize_fixed());
+        let z = <Scalar as Reduce<U256>>::from_be_bytes_reduced(msg_digest.finalize());
```

Comment:
If the `digest` feature of the `elliptic-curve` crate is enabled, you can now use the following:

```suggestion
        let z = <Scalar as Reduce<U256>>::from_be_digest_reduced(msg_digest);
```

Reply:
(this comment applies to `ecdsa` too but I guess I missed it there)

Reply:
I guess this requires RustCrypto/traits#904.

Reply:
See RustCrypto/signatures#439.

Reply:
Yeah, the bounds were bad. May need to cut another release of `elliptic-curve`.

Reply:
Can do another PR when the new release is ready.

Reply:
Or just push in here actually.

Reply:
Done.